tools/src/wyvern/tools/interop/LazyStructuralType.java

```diff
@@ -17,11 +20,13 @@
     private Class<?> javaClass;
     private TypeContext ctx;
     private boolean initialized = false;
+    private boolean safe = true;
 
-    public LazyStructuralType(Class<?> javaClass, TypeContext ctx) {
+    public LazyStructuralType(Class<?> javaClass, TypeContext ctx, boolean safe) {
```

Comment:
Also add some javadoc comments in this class about what the constructor parameter and field "safe" means